rosserial_arduino/src/rosserial_arduino/make_libraries_isolated.py

```diff
@@ -89,6 +89,11 @@
 microcontroller_ws = parsed_contents["microcontroller_ws"]
 header_out = os.path.join(microcontroller_ws, parsed_contents["header_output_directory"])
 source_out = os.path.join(microcontroller_ws, parsed_contents["source_output_directory"])
+
+microcontroller_ws = os.path.expanduser(microcontroller_ws)
+header_out = os.path.expanduser(header_out)
+source_out = os.path.expanduser(source_out)
+
 custom_ws = parsed_contents["catkin_workspaces"]
 packages = parsed_contents["packages"]
```

Comment:
@bhavyangupta You need to `os.path.expanduser` for the `catkin_workspaces` as well ⚠️ Either here or when you iterate over the contents of `catkin_workspaces` (L109)